sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr_properties.asciidoc

```diff
@@ -0,0 +1,158 @@
+= sycl_ext_oneapi_annotated_ptr_properties
```

Comment:
I don't think it makes sense to have a separate extension named "sycl_ext_oneapi_annotated_ptr_properties".  If we have "generic" properties for `annotated_ptr` (e.g. `align`), they should go directly in the "sycl_ext_oneapi_annotated_ptr" extension.

It's OK to have even more properties that are defined in other extensions.  For example, maybe we want a separate extension for `buffer_location`.  However, I don't see the benefit of having a separate extension for generic properties.  That just adds one more extension and feature-test macro that users need to uderstand.

Reply:
- Moved the basic properties : `alignment`, `runtime_aligned`, and `kernel_args_restrict` into this document.
- Renamed the other document to `sycl_ext_intel_fpga_annotated_ptr_properties`
- Added the additional properties we want for FPGA devices to the new document.

The properties in `sycl_ext_intel_fpga_annotated_ptr_properties` were going to be added to `annotated_argument` but our current model is to put all pointer properties on `annotated_ptr` and clarify in property definition that these need only be retained on kernel argument and not within device code. Update the implicit conversion operator description reflect this.

The name `buffer_location` is already being used by the runtime property, I have used the name `mem_id` in the new document. Is that okay? Or can I reuse the name `buffer_location` here because these extensions define compile-time constant properties? Either way we wanted a better name than `buffer_location`, hence I used `mem_id` which was floated earlier and was deemed acceptable.

+ @GarveyJoe

Reply:
I am continuing to use the `buffer_location` name.